File: dailyQuestion/2020-06/06-02/python/solution.py

def lengthOfLongestSubstring(s: str) -> int:
    tmp = set()
    result = 0
    j = 0
    n = len(s)
    for i in range(n) :
        if i != 0 :
            tmp.remove(s[i-1])
        while j < n and s[j] not in tmp :
            tmp.add(s[j])
            j += 1
        result = max(result, j - i)
    return result

if __name__ == "__main__" :
    a = [23,4,5,6]

    s = "abcabcbb"
    result = lengthOfLongestSubstring(s)
    print(result)

File: dailyQuestion/2020-06/06-02/golang/solution.go
package main

import (
	"fmt"
)

func main() {

	s := "abcabcbb"
	result := lengthOfLongestSubstring(s)
	fmt.Println(result)

}

func lengthOfLongestSubstring(s string) int {
	tmp := map[byte]int{}
	j := 0
	result := 0
	n := len(s)
	for i := 0; i < n; i++ {
		if i != 0 {
			delete(tmp, s[i-1])
		}
		for j < n && tmp[s[j]] == 0 {
			tmp[s[j]]++
			j += 1
		}
		result = max(result, j-i)
	}
	return result
}

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}
